lib/ocpp/v201/connectivity_manager.cpp

```diff
@@ -85,198 +88,141 @@ ConnectivityManager::get_network_connection_profile(const int32_t configuration_
         std::string::npos) {
         EVLOG_AND_THROW(std::runtime_error("ChargePointId must not contain \':\'"));
     }
 
-    // cache the network profiles on initialization
-    if (!cache_network_connection_profiles()) {
-        EVLOG_warning << "No network connection profiles configured, aborting websocket connection.";
-        return;
-    }
+    // Check the cache runtime since security profile might change async
+    this->check_cache_for_invalid_security_profiles();
 
-    const int config_slot_int = this->get_active_network_configuration_slot();
-
-    const auto network_connection_profile = this->get_network_connection_profile(config_slot_int);
+    const int configuration_slot_to_set =
+        this->pending_configuration_slot.value_or(this->get_active_network_configuration_slot());
+    const auto network_connection_profile = this->get_network_connection_profile(configuration_slot_to_set);
     // Not const as the iface member can be set by the configure network connection profile callback
-    auto connection_options = this->get_ws_connection_options(config_slot_int);
+    auto connection_options = this->get_ws_connection_options(configuration_slot_to_set);
     bool can_use_connection_profile = true;
 
     if (!network_connection_profile.has_value()) {
-        EVLOG_warning << "No network connection profile configured for " << config_slot_int;
+        EVLOG_warning << "No network connection profile configured for " << configuration_slot_to_set;
+        can_use_connection_profile = false;
+    } else if (!connection_options.has_value()) {
+        EVLOG_warning << "Connection profile configured for " << configuration_slot_to_set << " failed: not valid URL";
         can_use_connection_profile = false;
     } else if (this->configure_network_connection_profile_callback.has_value()) {
-        EVLOG_debug << "Request to configure network connection profile " << config_slot_int;
-
-        std::future<ConfigNetworkResult> config_status = this->configure_network_connection_profile_callback.value()(
-            config_slot_int, network_connection_profile.value());
-        const int32_t config_timeout =
-            this->device_model.get_optional_value<int>(ControllerComponentVariables::NetworkConfigTimeout)
-                .value_or(default_network_config_timeout_seconds);
-
-        std::future_status status = config_status.wait_for(std::chrono::seconds(config_timeout));
-
-        switch (status) {
-        case std::future_status::deferred:
-        case std::future_status::timeout: {
-            EVLOG_warning << "Timeout configuring config slot: " << config_slot_int;
+        if (std::optional<std::string> interface_address =
+                get_network_configuration_interface(configuration_slot_to_set, network_connection_profile.value());
+            interface_address.has_value()) {
+            connection_options->iface = interface_address;
+        } else {
             can_use_connection_profile = false;
```

Comment:
Id prefer to add a log that mentions that the profile for this slot can not be used.